libcloudforensics/aws.py

```diff
@@ -0,0 +1,705 @@
+
+
+class AWSSnapshot(AWSElasticBlockStore):
+  """Class representing an AWS EBS snapshot.
+
+  Attributes:
+    snapshot_id (str): The id of the snapshot.
+    volume (AWSVolume): The volume from which the snapshot was taken.
+    name (str): The name tag (if any) of the snapshot.
+  """
+  def __init__(self, snapshot_id, volume, name=None):
+    """Initialize an AWS EBS snapshot.
+
+    Attributes:
+      snapshot_id (str): The id of the snapshot.
+      volume (AWSVolume): The volume from which the snapshot was taken.
+      name (str): The name tag (if any) of the snapshot.
+    """
+    super(AWSSnapshot, self).__init__(volume.aws_account, volume.region,
+                                      volume.availability_zone, name)
+    self.snapshot_id = snapshot_id
+    self.volume = volume
+
+  def Delete(self):
+    """Delete a snapshot."""
+    client = self.aws_account.ClientApi(EC2_SERVICE)
+    try:
+      client.delete_snapshot(
+          SnapshotId=self.snapshot_id)
+    except client.exceptions.ClientError as exception:
+      raise RuntimeError('Error: could not delete snapshot {0:s}: {1:s}'.format(
+          self.snapshot_id, str(exception)))
+
+
+def CreateVolumeCopy(instance_name, zone, volume_name=None):
+  """Create a copy of an AWS EBS Volume.
+
+  Attributes:
+    instance_name (str): Instance using the volume to be copied.
+    zone (str): The zone within the region to create the new resource in.
+    volume_name (str): Name of the volume to copy. If None, boot volume will be
+        copied.
+
+  Returns:
+    AWSVolume: An AWS EBS Volume object.
+
+  Raises:
+    RuntimeError: If there are errors copying the volume.
+  """
+  aws_account = AWSAccount(zone)
+  instance = aws_account.GetInstance(instance_name) if instance_name else None
+
+  try:
+    if volume_name:
+      volume_to_copy = aws_account.GetVolume(volume_name)
+    else:
+      volume_to_copy = instance.GetBootVolume()
+
+    log.info('Volume copy of {0:s} started...'.format(volume_to_copy.volume_id))
+    snapshot = volume_to_copy.Snapshot()
+    new_volume = aws_account.CreateVolumeFromSnapshot(
+        snapshot, volume_name_prefix='evidence'
+    )
+    snapshot.Delete()
+    log.info('Volume {0:s} successfully copied to {1:s}'.format(
+        volume_to_copy.volume_id, new_volume.volume_id))
+
+  except RuntimeError as exception:
+    error_msg = 'Error copying volume {0:s}: {1!s}'.format(
+        volume_name, exception)
+    raise RuntimeError(error_msg)
+
+  return new_volume
+
+
+def GetTagForResourceType(resource, name):
+  """Create a dictionary for AWS Tag Specifications.
+
+  Attributes:
+    resource (str): The type of AWS resource.
+    name (str): The name of the resource.
+
+  Returns:
+    dict: A dictionary for AWS Tag Specifications.
+  """
+  return dict(
+      ResourceType=resource,
+      Tags=[
+          dict(
+              Key='Name',
+              Value=name
+          )
+      ]
+  )
```

Comment:
Using dict litterals, this would look like

```python
return {
  'ResourceType': resource,
  'Tags': [{
    'Key': 'Name',
    'Value': name,
  }],
}
```

which IMO is much nicer to read.